SENG-499-Company2-B01/Backend
# Sprint 1 PR

# Sprint 1 PR 
This PR contains all the work done for sprint 1. In this sprint we created a basic golang web-server and a mongo database which contains the main schedule_db for the overall application. Within the database there are 4 collections for users, courses, classrooms, and schedules. The web-server contains all the CRUD API endpoints for users, courses, classrooms, and schedules. The module directory contains the implementation for each of the 4 main endpoints (users, ccourses, classrooms, schedules).   

## Steps to test / demo 

1. Start up the backend: 
`docker-compose up --build -d` 

2. Verify the all containers are up and healthy: 
`docker ps -a` 

3. Verify the DB is properly intialized via mongo compass or the following: 
`docker exec -it backend_mongodb_1 bash`, then run `mongo -u <username> -p <password>` 
then run `show collections`, (verify schedule_db exists)  `use schedule_db`, finally run `show collections` 

4. Use postman to verify CRUD APIs are working properly
